pkg/hcl_interpreter/hcl_interpreter.go

```diff
@@ -245,6 +245,16 @@ func (v *Evaluation) evaluate() error {
 
 		vars := v.prepareVariables(name, term)
 		vars = MergeVal(vars, NestVal(LocalName{"path", "module"}, cty.StringVal(moduleMeta.Dir)))
+		vars = MergeVal(vars, NestVal(LocalName{"path", "root"}, cty.StringVal(v.Analysis.Modules[""].Dir)))
```

Comment:
From dumping serialisations of Analysis.Modules, we appear to use the empty string as the key for the root module. I have half-convinced myself that this key will always exist, e.g. https://github.com/snyk/policy-engine/blob/f4985f279b1c284404dd28e9e3b4aa3225f9e30d/pkg/hcl_interpreter/names.go#L45C44-L45C44 will always return empty string at least once per scan.

I wonder if it's worth being defensive to avoid a panic here.

Reply:
I would use `ModuleNameToString(EmptyModuleName)` here instead of `””`.

IMO it’s worth adding a branch to avoid the panic, even though it won’t trigger right now, there could be a future in which we produce an analysis without modules…

Reply:
Good call @jaspervdj-snyk. I've updated the PR to only set path.root if the expected root module is not nil. Is that what you had in mind, or perhaps we pass in some sensible stub for it, like the value of path.module?